### ISSUE TITLE
Download component from source url workflow can't find file to copy

### ISSUE THREAD

**Issue by anitacaron** [INCATools/ontology-development-kit#1060]
When the component is different, it copies from the tmp folder, but when it's downloading for the first time, it shouldn't try to copy.

Error on test [PR](https://github.com/obophenotype/uberon/pull/2975) https://github.com/obophenotype/uberon/actions/runs/8389532879/job/22975944727

![Screenshot 2024-05-17 at 11 36 07](https://github.com/INCATools/ontology-development-kit/assets/2208124/a8786f3a-b4b5-4c06-902a-4d89757d9514)

**Comment by anitacaron** [comment]
I found out that the issue is in the QC. Now, we need to set `MIR=false`, and to download the component, the condition is `MIR=true`.

https://github.com/INCATools/ontology-development-kit/blob/dbcab69c7d8ce1226522e4ea143a7a00ab156308/template/src/ontology/Makefile.jinja2#L615-L618

What are your thoughts? @matentzn @gouttegd

**Comment by gouttegd** [comment]
I think that deep down, it’s all a consequence of the old approach of using shell conditionals instead of Make conditionals (see #912). Make is unaware of shell conditionals, so when you have a rule like this:

```Make
component-download-{{ component.filename }}: | $(TMPDIR) 
 	if [ $(MIR) = true ] && [ $(COMP) = true ]; then ...ALL THE CODE OF THE RULE... ; fi
```

when the entirety of what the rule does is within a shell conditional, when the condition is false Make will have no idea that the rule didn’t actually do anything. From Make’s point of view, the shell code has been executed, that’s all that matters.

We should convert all those shell conditionals to Make conditionals as we recently did for the mirroring rules (#973).

Something like that:

```Make

ifeq ($(COMP),true)
# COMP=true -> we regenerate the components

ifeq ($(MIR),true)
# MIR=true -> even the components that are mirrored from a foreign source

.PHONY: component-download-hra_subset.owl
component-download-hra_subset.owl: | $(TMPDIR)
	$(ROBOT) merge -I https://raw.githubusercontent.com/hubmapconsortium/ccf-validation-tools/master/owl/UB_ASCTB_subset.owl \
	annotate --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) -o $(TMPDIR)/$@.owl

$(COMPONENTSDIR)/hra_subset.owl: component-download-hra_subset.owl
	if cmp -s $(TMPDIR)/component-download-hra_subset.owl.owl $(TMPDIR)/component-download-hra_subset.owl.tmp.owl ; then echo "Component identical."; \
      else echo "Component is different, updating." &&\
		cp $(TMPDIR)/component-download-hra_subset.owl.owl $(TMPDIR)/component-download-hra_subset.owl.tmp.owl &&\
		$(ROBOT) annotate -i $(TMPDIR)/component-download-hra_subset.owl.owl --ontology-iri $(ONTBASE)/$@ $(ANNOTATE_ONTOLOGY_VERSION) -o $@; fi

.PRECIOUS: $(COMPONENTSDIR)/hra_subset.owl

endif

# Here, we can re-generate the other components (those that don't depend on a foreign source, and therefore that can be built even under MIR=false)

else

# COMP=false -> Not regenerating any component; if any rule depends on a component, Make would simply notice that the component files already exist in `$(COMPONENTSDIR)` (assuming the components are committed to the repository, which they normally should be), and will not do anything.

endif
```

**Comment by anitacaron** [comment]
Yeah, but still, if MIR=false, it won't download the component, right?

**Comment by gouttegd** [comment]
No indeed, but isn’t that what is wanted here? We only want to refresh that component when MIR=true.

**Comment by anitacaron** [comment]
I'll create the PR, thanks.

**Comment by gouttegd** [comment]
> Actually, the second goal only checks for COMP=true and not the MIR

Yeah, but since the rule that this one depends on does also check `MIR`, this does not make any sense. _In effect_, and unless we happen to have a `$(TMPDIR)/component-download-{{ component.filename }}.owl` file that would be a leftover from a previous run (the kind of things we should definitely not rely upon), we can only regenerate that component when both `MIR` and `COMP` are true. I assume this is the desired behaviour.

**Comment by gouttegd** [comment]
This is actually already fixed since #1168, which replaced the use of shell conditionals by Make conditionals (in effect implementing the solution proposed in [this comment](https://github.com/INCATools/ontology-development-kit/issues/1060#issuecomment-2117586763)).
